tools/ci/src/commands/compile_check_no_std.rs

```diff
@@ -102,6 +102,14 @@ impl Prepare for CompileCheckNoStdCommand {
             "Please fix compiler errors in output above for bevy_ecs no_std compatibility.",
         ));
 
+        commands.push(PreparedCommand::new::<Self>(
+            cmd!(
+                sh,
+                "cargo check -p bevy_app --no-default-features --features bevy_reflect --target {target}"
```

Comment:
Now that `bevy_app` is compatible, we can start looking into replacing/augmenting this `compile_check_no_std` CI task with something more robust and tangible, like a `no_std` example crate.